### DISCUSSION TITLE
"isaddress":false problem

### DISCUSSION THREAD

**Discussion by dlubom** [osm-search/Nominatim#3302] [General]
Hello, I have question: why in this query, for hotel in Rabat, city is marked as isaddress=false for city relation ?
API link: https://nominatim.openstreetmap.org/details.php?osmtype=N&osmid=2430191199&class=tourism&addressdetails=1&hierarchy=0&group_hierarchy=1&format=json

Nominatim link: https://nominatim.openstreetmap.org/ui/details.html?osmtype=N&osmid=2430191199&class=tourism

![image](https://github.com/osm-search/Nominatim/assets/5931841/4e5a9703-f509-4f8e-9580-0f0da9efe577)
Is it bug ?

Second question, can I somehow get in this query https://nominatim.openstreetmap.org/details.php?osmtype=N&osmid=2430191199&class=tourism&addressdetails=1&hierarchy=0&group_hierarchy=1&format=json
not only chain of relation but also connected to relation places ?

**Comment by mtmail** [comment]
The HTML output is easier to read:

https://nominatim.openstreetmap.org/ui/details.html?osmtype=N&osmid=2430191199&class=tourism&addressdetails=1&hierarchy=0&group_hierarchy=1

`is_address` mean the place was considered to be part of the address (the address table) but another place was choosen. In this case "Agdal Riyad" was the "winner" for the admin_level=8 (city). With both admin_level=8 relations Nominatim needs to make a decision if https://www.openstreetmap.org/relation/2799215 or https://www.openstreetmap.org/relation/2799211 is more relevant. They cover the same area. I would say "Agdal Riyad" as arrondissement has the wrong admin_level set but a local mapper would need to determine that.

I'm not sure I understand the second question. When a relation and a node represent the same place then Nominatim merges the together internally. For example city of Rabat is https://www.openstreetmap.org/relation/2799215 and https://www.openstreetmap.org/node/299120862 Here Nominatim created https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=2799215&class=boundary which has tags from both. It has the `place=city` from the node and `admin_level=8` from the relation. Nominatim stores a `linked_place_id` value from the node to the relation but not from relation to the node:

```
select linked_place_id from placex where osm_type='N' and osm_id=299120862;
 linked_place_id
-----------------
       261186595

select osm_type, osm_id from placex where place_id = 261186595;
 osm_type | osm_id
----------+---------
 R        | 2799215
```

**Comment by dlubom** [comment]
Thank you, I send priv msg to author of this change: https://www.openstreetmap.org/relation/2799211/history

Now, lets come back to second question:

Let's phrase the question differently, more generally: I intend to periodically query Nominatim for various places, including the cities/states/countries associated with them. This will result in many requests, so I am considering setting up my own server.

Here's the question: do you see a possibility to automatically detect and correct errors similar to the one mentioned above on the fly, so as to obtain, despite the error, the city of Rabat for the above-mentioned hotel?

If this possibility requires code modification, that's fine with me.

**Reply by mtmail** [reply]
The /details endpoint is not allowed to be used by scripts or bots (https://nominatim.org/release-docs/latest/api/Details/), you need to setup your own server.
To find overlapping relations of the same admin_level I would use the Overpass API. It displays found areas transparent, if two overlap the color will be darker. 

```
[out:json][timeout:100];
area["ISO3166-1"="MA"]->.country;
(
  relation["admin_level"="8"](area.country);
);
out geom;
```
![image](https://github.com/osm-search/Nominatim/assets/3727288/2ab681b3-e6c4-4297-a7f6-4a1b737684a3)
